packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1616,15 +1620,32 @@ class _SelectableFragment with Selectable, ChangeNotifier implements TextLayoutM
     return result;
   }
 
-  TextPosition _getNextPosition(TextBoundary boundary, TextPosition position, bool forward) {
-    if (forward) {
-      return _clampTextPosition(
-        (PushTextPosition.forward + boundary).getTrailingTextBoundaryAt(position)
-      );
+  TextPosition _beyondTextBoundary(TextPosition extent, bool forward, TextBoundary textBoundary) {
```

Comment:
extent means the selection end. maybe rename the parameter back to position or offset?

Reply:
also the _beyondTextBoundary seems like a weird name...

Reply:
renamed